aws/aws-sdk-go-v2
# Upgrade to smithy 1.25.0

# Dependent on Smithy 1.25.0 release and aws/smithy-go#386, BLOCK merging until then.

For changes to files under the `/model/` folder, and manual edits to autogenerated code (e.g. `/service/s3/api.go`) please create an Issue instead of a PR for those type of changes.

N/A

---

If there is an existing bug or feature this PR is answers please reference it here.

N/A

---

*Description of changes:*

**Note: This change introduces break/fix's for pointable types.**

Update EC2 unboxed shapes to boxed shapes backfill customization

Due to changes in smithy 2.0 nullability semantics around the `@box` trait, the generated code for `ec2.json` generated incorrect code.

Instead of adding `@box` traits to certain members in the customization, the `@clientOptional` trait is added to members of all structure shapes.

Chores:

- Add mavenLocal to codegen build files
- Update .gitignore to ignore .idea/

Regenerate clients with break/fix's

Due to prior bugs for `@box` members not being serialized as a pointer type and incorrect pagination serialization, the clients in this commit are regenerated with the break/fix's.

---

*Testing:*

- Use smithy changes from awslabs/smithy#1394
  - `./gradlew clean build publishToMavenLocal smithy-cli:runtime`
- Build smithy-go with (aws/smithy-go#386) and smithy (awslabs/smithy#1394) changes: `make smithy-clean smithy-build smithy-publish-local smithy-clean`
- Build aws-sdk-go-v2 with current changes and smithy-go (aws/smithy-go#386) and smithy (awslabs/smithy#1394) changes: `make generate` and `make unit`
  - Verify [expected diff](https://github.com/syall/aws-sdk-go-v2/blob/nullability-fix-diffs/aws-sdk-go-v2-diff.diff) (should be a lot of break/fix's for pointable types that were serialized as value types before).

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
